src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5942,9 +5942,9 @@ private bool IsUsingAliasInScope(string name)
                     options |= LookupOptions.MustBeInvocableIfMember;
                 }
 
-                var typeArgumentsSyntax = right.Kind() == SyntaxKind.GenericName ? ((GenericNameSyntax)right).TypeArgumentList.Arguments : default(SeparatedSyntaxList<TypeSyntax>);
+                SeparatedSyntaxList<TypeSyntax> typeArgumentsSyntax = right is GenericNameSyntax genericName ? genericName.TypeArgumentList.Arguments : default;
```

Comment:
>SeparatedSyntaxList<TypeSyntax> typeArgumentsSyntax = right is GenericNameSyntax genericName ? genericName.TypeArgumentList.Arguments : default; [](start = 16, length = 144)

Is there a change in behavior in this function? If not, please revert the refactoring.